From d234caf0375056269718eae7de316407a0144f37 Mon Sep 17 00:00:00 2001
From: Li Jun <jun.li@nxp.com>
Date: Tue, 26 Jun 2018 18:16:48 +0800
Subject: [PATCH 4119/5242] MLK-17921-3 driver: irqchip: shorten wakeup wait
 time of IPI SW workaround

commit  e464773dd649b47b97bbbff794ad083bd72c1e9a from
https://source.codeaurora.org/external/imx/linux-imx.git

Shorten the delay time from 50us to be 10us introduced by
commit 58cef5f23bd0 ("MLK-16804-04 driver: irqchip: Add IPI SW workaround for imx8mq"),
tested various basic cases and didn't find issues.

CC: Bai Ping <ping.bai@nxp.com>
Acked-by: Peter Chen <peter.chen@nxp.com>
Signed-off-by: Li Jun <jun.li@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/irqchip/irq-gic-v3.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/irqchip/irq-gic-v3.c b/drivers/irqchip/irq-gic-v3.c
index 3ec9eb9..7130fda 100644
--- a/drivers/irqchip/irq-gic-v3.c
+++ b/drivers/irqchip/irq-gic-v3.c
@@ -767,7 +767,7 @@ static void gic_raise_softirq(const struct cpumask *mask, unsigned int irq)
 			val |= (1 << 12);
 			writel_relaxed(val, iomuxc_gpr_base + 0x4);
 			/* delay for a while to make sure cores wakeup done */
-			udelay(50);
+			udelay(10);
 			val &= ~(1 << 12);
 			writel_relaxed(val, iomuxc_gpr_base + 0x4);
 		}
-- 
1.7.9.5

